### ISSUE TITLE
Rust plugin does not autocomplete or go to dependencies unless Cargo.toml is at the root of the project

### ISSUE THREAD

**Issue by egorklepikov** [intellij-rust/intellij-rust#9307]
From https://youtrack.jetbrains.com/issue/IDEA-301218

**tl;dr**

The Rust plugin does not work well unless Cargo.toml is at the root of the project.

**Desired Behavior**

The Rust plugin is able to work in a repo with Go, TypeScript, and Rust source code when that Rust source code and the Cargo.toml file are not at the root of the project.

**Actual Behavior**

The Rust plugin only works as expected when the Cargo.toml file is at the root of the project.

**Background**

My project is a monorepo with some Go, TypeScript, and now Rust. After installing dependencies through cargo, IntelliJ is unable to go to definitions for my Rust dependencies -- those I've written or those downloaded from Cargo -- or offer valid autocompletion when the with my Cargo.toml file is nested. I keep getting an error banner "No cargo projects found" and when I attempt to attach to my Cargo file it complains that file is already attached to another workspace -- despite me only having one file.

**Example Directory Structure**

```
/docs
/cmd (with Go files)
/pkg (with Go files)
/ui (TypeScript/SolidsJS)
/services/
/services/data
/services/data/Cargo.toml
/services/data/src/main.rs (this file seems ok)
/services/data/src/api/api.rs (this file does not have full features of autocomplete, go to dependencies, etc.)
```

IU-222.3345.118, JRE 17.0.3+7-b469.32x64 JetBrains s.r.o., OS Mac OS X(x86_64) v12.5.1, screens 4096.0x2560.0; Retina

**Comment by vlad20012** [comment]
Hello!
The case of Cargo.toml outside of a project root is perfectly supported, so you're likely facing another bug or misconfiguration.

> it complains that file is already attached to another workspace

If it complains, then, well, it can really be attached.
Have you tried to open the Cargo tool window and see which Cargo projects are linked there? Could you please make a sceenshot?
![image](https://user-images.githubusercontent.com/3221931/188438492-c1dbe03f-f462-4b30-8a3a-2972f924923d.png)

**Comment by aarongreenlee** [comment]
This was created on my behalf after filing https://youtrack.jetbrains.com/issue/IDEA-301218. 

I've deleted the Rust source code from my project and set everything up again fresh -- and now I see the desired behavior. While I think we can close this as my issue is not currently reproducible I can share this has happened to me on a few occasions. 

Feel free to close and I'll post back if I can reproduce.

With gratitude,

Aaron
